com.playeveryware.eos/Runtime/Core/Common/SetOfNamed.cs

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.Common
+{
+    using System.Collections.Generic;
+    using System;
+
+    /// <summary>
+    /// Class that stores a unique set of values, where each value has a name
+    /// associated to it. Items are sorted by the name. No two items of the same
+    /// value can be added (regardless of if they have different names). Also,
+    /// no two items can have the same name (regardless of if they have
+    /// different values).
+    /// </summary>
+    /// <typeparam name="T">The type being wrapped with a name.</typeparam>
+    public class SetOfNamed<T> : HashSet<Named<T>> where T : IEquatable<T>
```

Comment:
Would be good to collect input on whether this is a code smell (inheritance vs composition).

Reply:
Input collected - consensus was that this is an acceptable use of inheritance. Consequently, I am closing this conversation.